tooling/miniutil/src/run.rs

```diff
@@ -0,0 +1,33 @@
+use crate::*;
+
+use gen_minirust::prelude::NdResult;
+use std::collections::HashSet;
+
+// Run the program and return its TerminationInfo.
+// We fix `BasicMemory` as a memory for now.
+pub fn run_program(prog: Program) -> TerminationInfo {
+    let res: NdResult<!> = try {
+        let mut machine = Machine::<BasicMemory>::new(prog)?;
+
+        loop {
+            machine.step()?;
+            mark_and_sweep(&machine);
+        }
+    };
+
+    // Extract the TerminationInfo from the `NdResult<!>`.
+    let res: Result<!, TerminationInfo> = res.get_internal();
+    match res {
+        Ok(never) => never,
+        Err(t) => t,
+    }
+}
+
+// This drops everything not reachable from the machine.
+fn mark_and_sweep<M: Memory>(machine: &Machine<M>) {
+    // `set` is the set of gc pointers directly pointed to by `machine`.
+    let mut set = HashSet::new();
+    machine.points_to(&mut set);
+
+    gen_minirust::libspecr::hidden::mark_and_sweep(set);
```

Comment:
A nicer API might be to pass `machine` to `mark_and_sweep`? Basically that function could take any parameter that has a `points_to` function, then the caller doesn't have to do this.

Reply:
I originally just received a single `<T: GcCompat>` there, but in the general case you might want to have multiple roots.
So to keep it flexible, I went for a `HashSet`.

I could change it to `&[dyn GcCompat]` or something similar though

Reply:
This is the only user of that API, it doesn't seem worth to over-generalize. But if you really want, then maybe have a general `mark_and_sweep1 that takes a list and a little helper (still in the gc lib) that works for a single root?

Reply:
Okay, I'll change the API back to `fn mark_and_sweep(root: impl GcCompat)`.
new libspecr and specr-transpile version is on the way